components/list/demo/list.html

```diff
@@ -0,0 +1,221 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta http-equiv="X-UA-Compatible" content="ie=edge">
+		<meta charset="UTF-8">
+		<link rel="stylesheet" href="../../demo/styles.css" type="text/css">
+		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+		<script type="module">
+			import '../../demo/demo-page.js';
+			import '../../typography/typography.js';
+			import '../list.js';
+			import './list-demo-name-image.js';
+			import './list-demo-course-image.js';
+			import '../../button/button-icon.js';
+		</script>
+	</head>
+	<body unresolved>
+
+		<d2l-demo-page page-title="List Components">
+			<h2>Basic List</h2>
+			<d2l-demo-snippet>
+				<d2l-list>
+					<d2l-list-item>Item 1</d2l-list-item>
+					<d2l-list-item>Item 2</d2l-list-item>
+					<d2l-list-item>Item 3</d2l-list-item>
+				</d2l-list>
+			</d2l-demo-snippet>
+			<d2l-demo-snippet>
+				<d2l-list divider="middle">
+					<d2l-list-item>
+						<div class="d2l-list-item-text d2l-body-compact">Identify categories of physical activities</div>
+						<div class="d2l-list-item-text-secondary d2l-body-small">Specific Expectation A1.2</div>
+					</d2l-list-item>
+					<d2l-list-item>
+						<div class="d2l-list-item-text d2l-body-compact">Apply a decision-making process to assess risks and make safe decisions in a variety of situations</div>
+						<div class="d2l-list-item-text-secondary d2l-body-small">Specific Expectation B2.1</div>
+					</d2l-list-item>
+					<d2l-list-item>
+						<div class="d2l-list-item-text d2l-body-compact">Retain objects of various shapes and sizes in different ways, while moving around others and equipment</div>
+						<div class="d2l-list-item-text-secondary d2l-body-small">Specific Expectation B2.2</div>
+					</d2l-list-item>
+				</d2l-list>
+			</d2l-demo-snippet>
+
+			<h2>List of Users</h2>
+
+			<d2l-demo-snippet>
+				<d2l-list>
+					<d2l-list-item>
+						<d2l-list-demo-name-image color="purple" letters="BN" slot="illustration"></d2l-list-demo-name-image>
```

Comment:
I couldn't figure out a nice way to get those to work with hovering. I did start with making `d2l-list-item-content` and didn't find that it was required.

For the primary action it will be easier if it is apart of `<d2l-list-item href="...">` if href is there then it is clickable. Easier to add hovering.

This also made the inner and outer illustrations fall out which was a slight bonus.